### PR TITLE
use taskQueueId internally to the queue (functions and code for tasks)

### LINKED ISSUE #3578: use taskQueueId internally to the queue
This bug covers refactoring the queue to use taskQueueId internally, converting back and forth to provisionerId/workerType as necessary to keep external interfaces unchanged.

### PR DESCRIPTION
**Pull request by alopezz**
Github Bug/Issue: Fixes #3578 

With #3987, the db will already store `task_queue_id`s rather than `provisioner_id` and `worker_type` for tasks. This PR implements the follow up of making the code also do this internally (while keeping external interfaces unchanged).

### CODE REVIEW

File: db/versions/0061.yml [hunk 1]
```diff
@@ -0,0 +1,395 @@
+version: 61
+description: Deprecate functions that use provisioner_id/worker_type and add equivalents using task_queue_id instead, for tasks table
+methods:
+  create_task:
```
Comment (djmitche) [inline db/versions/0061.yml:4]: From what I can see, this is the same function from version 0060, but with `deprecated: true`.  As a shortcut, you can omit everything except `deprecated` for such changes.

**Review by djmitche — CHANGES_REQUESTED** [review]
Aside from the below minor detail, this looks great!

**Review by djmitche — APPROVED** [review]
Looks good once it's green!  I pushed a `yarn generate`.

**Comment by alopezz** [comment]
Yay! #3578 is finally really closed!

**Pull request merged**